lib/sql-logging/configuration.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+module SqlLogging
+  class Configuration
+    DEFAULTS = {
+      show_sql_backtrace: true,
+      show_top_sql_queries: :total_time,
+      top_sql_queries: 10,
+      backtrace_cleaner: nil,
+      logger: nil
+    }.freeze
+
+    attr_accessor(*DEFAULTS.keys)
+
+    def initialize
+      DEFAULTS.each do |k, v|
+        instance_variable_set("@#{k}", v)
+      end
+    end
+
+    def show_top_sql_queries=(value)
+      validate_allowed([false, :rows, :queries, :bytes, :total_time,
+                        :median_time], value, __method__)
```

Comment:
Thoughts on using `caller_locations` from within the method so it does not have to pollute the call?